Plugins/Effect/Effect.cpp

```diff
@@ -285,4 +306,142 @@ ArgumentStack Effect::ReplaceEffect(ArgumentStack&& args)
+            eff->m_sParamString[3] = Services::Events::ExtractArgument<std::string>(args).c_str();
+            eff->m_sParamString[2] = Services::Events::ExtractArgument<std::string>(args).c_str();
+            eff->m_sParamString[1] = Services::Events::ExtractArgument<std::string>(args).c_str();
+            eff->m_sParamString[0] = Services::Events::ExtractArgument<std::string>(args).c_str();
+
+            eff->m_nParamFloat[3] = Services::Events::ExtractArgument<float>(args);
+            eff->m_nParamFloat[2] = Services::Events::ExtractArgument<float>(args);
+            eff->m_nParamFloat[1] = Services::Events::ExtractArgument<float>(args);
+            eff->m_nParamFloat[0] = Services::Events::ExtractArgument<float>(args);
+
+            eff->m_nParamInteger[7] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[6] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[5] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[4] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[3] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[2] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[1] = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nParamInteger[0] = Services::Events::ExtractArgument<int32_t>(args);
+
+            eff->m_nCasterLevel       = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_bShowIcon          = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_bExpose            = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nSpellId           = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_oidCreator         = Services::Events::ExtractArgument<ObjectID>(args);
+            eff->m_nExpiryTimeOfDay   = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_nExpiryCalendarDay = Services::Events::ExtractArgument<int32_t>(args);
+            eff->m_fDuration          = Services::Events::ExtractArgument<float>(args);
+            eff->m_nSubType           = Services::Events::ExtractArgument<int32_t>(args);
+        }
+    }
+
+    return Services::Events::Arguments();
+}
+
+ArgumentStack Effect::RemoveEffectById(ArgumentStack&& args)
+{
+    int32_t ret = 0;
+    auto objectId = Services::Events::ExtractArgument<ObjectID>(args);
+    if(objectId != Constants::OBJECT_INVALID)
+    {
+        if (auto *pObject = Utils::AsNWSObject(Globals::AppManager()->m_pServerExoApp->GetGameObject(objectId)))
+        {
+            auto id = Services::Events::ExtractArgument<std::string>(args);
+            auto it = std::find_if(id.begin(), id.end(), [](unsigned char c) { return !std::isdigit(c); });
+            if(!id.empty() && it == id.end())
+                ret = pObject->RemoveEffectById(std::stoi(id));
```

Comment:
`std::stoi` converts it to a 32bit integer, but the effect ID is 64bit. You want `std::stol()`.
Also, there is no need to sanity check it with `std::isdigit` because the conversion functions will throw an exception if the conversion doesn't go smoothly, which NWNX will already catch and print.